lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/DefaultJsonFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonFactoryBuilder;
+import io.airlift.json.ObjectMapperProvider;
+
+import static com.google.common.base.Verify.verify;
+
+public class DefaultJsonFactory
+{
+    public static final JsonFactory INSTANCE = new ObjectMapperProvider()
+            .get()
+            .getFactory();
```

Comment:
`ObjectMapperProvider` is focused on object mapping and configures that.
the fact that it disables string limits looks incidental from API perspective.
also, `ObjectMapperProvider` used for REST APIs probably should have some sane limits restored.

for internal query processing (eg json functions) we don't want limits. thus i think we should instantiate and configure `JsonFactory` directly, without calling upon `ObjectMapperProvider`